check_isa.cpp

```diff
@@ -115,36 +115,70 @@ static const char *lGetSystemISA() {
     // Call cpuid with eax=7, ecx=0
     __cpuidex(info2, 7, 0);
 
-    if ((info[2] & (1 << 27)) != 0 &&  // OSXSAVE
-        (info2[1] & (1 << 5)) != 0 &&  // AVX2
-        (info2[1] & (1 << 16)) != 0 && // AVX512 F
-        __os_has_avx512_support()) {
+    int info3[4];
+    // Call cpuid with eax=7, ecx=1
+    __cpuidex(info3, 7, 1);
+
+    // clang-format off
+    bool sse2 =                (info[3] & (1 << 26))  != 0;
+    bool sse4 =                (info[2] & (1 << 19))  != 0;
+    bool avx_f16c =            (info[2] & (1 << 29))  != 0;
+    bool avx_rdrand =          (info[2] & (1 << 30))  != 0;
+    bool osxsave =             (info[2] & (1 << 27))  != 0;
+    bool avx =                 (info[2] & (1 << 28))  != 0;
+    bool avx2 =                (info2[1] & (1 << 5))  != 0;
+    bool avx512_f =            (info2[1] & (1 << 16)) != 0;
+    bool avx512_dq =           (info2[1] & (1 << 17)) != 0;
+    bool avx512_pf =           (info2[1] & (1 << 26)) != 0;
+    bool avx512_er =           (info2[1] & (1 << 27)) != 0;
+    bool avx512_cd =           (info2[1] & (1 << 28)) != 0;
+    bool avx512_bw =           (info2[1] & (1 << 30)) != 0;
+    bool avx512_vl =           (info2[1] & (1 << 31)) != 0;
+    bool avx512_vbmi2 =        (info2[2] & (1 << 6))  != 0;
+    bool avx512_gfni =         (info2[2] & (1 << 8))  != 0;
+    bool avx512_vaes =         (info2[2] & (1 << 9))  != 0;
+    bool avx512_vpclmulqdq =   (info2[2] & (1 << 10)) != 0;
+    bool avx512_vnni =         (info2[2] & (1 << 11)) != 0;
+    bool avx512_bitalg =       (info2[2] & (1 << 12)) != 0;
+    bool avx512_vpopcntdq =    (info2[2] & (1 << 14)) != 0;
+    bool avx512_bf16 =         (info3[0] & (1 << 5))  != 0;
+    bool avx512_vp2intersect = (info2[3] & (1 << 8))  != 0;
```

Comment:
It should be on the line above before info3 starts.

Reply:
It's sorted on the "left" part, which is going in the order it's defined in ISA docs.